core/lib/zksync_core/src/api_server/web3/state.rs

```diff
@@ -543,10 +546,53 @@ impl<E> RpcState<E> {
 /// Contains mapping from index to `Filter` with optional location.
 #[derive(Default, Debug, Clone)]
 pub struct Filters {
-    state: HashMap<U256, TypedFilter>,
+    state: HashMap<U256, InstalledFilter>,
     max_cap: usize,
 }
 
+#[derive(Debug, Clone)]
+struct InstalledFilter {
+    pub filter: TypedFilter,
+    #[allow(dead_code)]
+    guard: Arc<GaugeGuard>,
```

Comment:
An `Arc` is a code smell, necessary only because of the `Clone` implementation. You can remove `derive(Clone)` for `InstalledFilter` and `Filters` since it's actually unused (which is natural – `Filters` are enclosed into an `Arc<RwLock<_>>`).

Reply:
true, I was more focused on making new structures work and not changing old ones to not break anything